content/en-us/reference/engine/classes/Debris.yaml

```
@@ -3,64 +3,61 @@ type: class
 category: Utility
 memory_category: Instances
 summary: |
-  The Debris service allows the developer to schedule the removal of the object
-  without yielding any code, through the usage of the `Class.Debris:AddItem()`
+  The Debris service allows the developer to schedule guaranteed destruction of
+  the object without yielding, through the usage of the `Class.Debris:AddItem()`
   method.
 description: |
-  The Debris service allows the developer to schedule the removal of the object
-  without yielding any code, through the usage of the `Class.Debris:AddItem()`
+  The **Debris** service allows scheduling of guaranteed destruction of an
+  object without yielding, through the usage of the `Class.Debris:AddItem()`
   method.
 
-  After the lifetime argument has elapsed (in seconds) the object is removed in
-  the same manner as `Class.Instance:Destroy()`.
-
   As Debris is a service it must be created using the
   `Class.ServiceProvider:GetService()` method.
 
-  **Why use Debris?**
+  #### Advantages
 
-  Beyond creating a bit of a mess, objects that are no longer required can use
-  up system memory and cause the game to run slower over time. For this reason
-  it is always advised to run the `Class.Instance:Destroy()` function on objects
-  you no longer need. However in many cases an object may have a specific period
-  of utility after which it needs to be destroyed.
+  Besides creating a bit of a mess, objects that are no longer required can use
+  up system memory and cause an experience to run slower over time. For this
+  reason, it's always advised to call `Class.Instance:Destroy()` on objects
+  you no longer need. In some cases, however, an object may have a specific
+  period of utility before it can be destroyed.
 
-  Take the example of projectile that has just been thrown. It could be cleaned
-  up using:
+  Consider a wall being smashed into individual bricks. If you want a
+  brick to linger for 3 seconds before being destroyed, you can use the
+  following code:
 
   ```lua
-  wait(3)
-  projectile:Destroy()
+  task.wait(3)
+  brick:Destroy()
   ```
 
-  However there are a number of issues with this approach. Firstly, it requires
-  yielding the code with a wait, which is not always desirable. Secondly, before
-  the 3 seconds have elapsed the object may have already been destroyed (for
-  example, if it reached `Class.Workspace.FallenPartsDestroyHeight`).
+  However, waiting causes the thread to yield which may not be desired. To avoid
+  yielding, a callback function can be scheduled to run on a new thread after 3
+  seconds:
 
   ```lua
-  delay(3, function()
-  	if projectile and projectile.Parent then
-  		projectile:Destroy()
-  	end
+  task.delay(3, function()
+  	brick:Destroy()
   end)
   ```
 
```

I took out the second suggestion of doing this "all on one line" because it seems that one "bad" example is enough. However, if you think it's useful to show both ways of calling task.delay(), feel free to add it back in.

---

I think it is useful to keep, because for some people, they might not know that there is a 1-line solution, which could be a downside to the reader compared to `task.delay`. Including both makes it clear what's happening with the multi-line solution, and clear that there is still a 1-line option if you want to use debris.

---

```suggestion

  Or in one line:
  
  ```lua
  task.delay(3, brick.Destroy, brick)
  ```

```